pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -179,6 +179,31 @@ func (e *timeSeriesQuery) processQuery(q *Query, ms *es.MultiSearchRequestBuilde
 	return nil
 }
 
+// Casts values to int when required by Elastic's query DSL
+func (metricAggregation MetricAgg) generateSettingsForDSL() map[string]interface{} {
+	setIntPath := func(path ...string) {
+		if stringValue, err := metricAggregation.Settings.GetPath(path...).String(); err == nil {
+			if value, err := strconv.Atoi(stringValue); err == nil {
+				metricAggregation.Settings.SetPath(path, value)
+			}
+		}
+	}
+
+	switch metricAggregation.Type {
+	case "moving_avg":
+		setIntPath("window")
+		setIntPath("predict")
+		setIntPath("settings", "alpha")
```

Comment:
Why alpha, beta, gamma is parse to int? To should be a float value 0-1

Reply:
Oh, just checked that it's the current behaviour in master (for dashboard queries and alerts), but isn't it a bug that could be addressed separately?

Reply:
that's true, i actually overlooked it! good catch!

Reply:
actually everything should be a float, not an int.

Reply:
> Also, how can we ensure that every time a new field is added this list is updated too?

my idea on the backend is that every type of aggregation will have a MarshalJson function, so that each of them can define only for itself what should be its own Elasticsearch DSL JSON representation. this is a short-term solution to make what we already have working.
makes sense?

Reply:
Coolio 👍 Makes sense. Before that happens - would it be worth leaving a comment with cross-reference between `public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/aggregations.ts` <->  `pkg/tsdb/elasticsearch/time_series_query.go:181` <-> `public/app/plugins/datasource/elasticsearch/query_builder.ts:344` as a reminder that if one is changed the others should be updated too?